### PR TITLE
Add test coverage

### PR DESCRIPTION
**Pull request by ewiggin**
## Description

Increase test coverage across generation dashboard components, form containers, service helpers, and API/customAxios layers.

This PR focuses on test-only work:
- adds dedicated specs for previously uncovered components and services
- stabilizes a few flaky coverage-only tests
- documents existing production quirks with FIXME comments where needed, without changing production behavior

## Checklist

Justify any unchecked point:

- [x] Changed code is covered by tests.
- [ ] Relevant changes are explained in the "Unreleased" section of the `CHANGES.md` file.
- [ ] That section includes "Upgrade notes" with any config, dependency or deploy tweek needed on development and server setups.
- [ ] Changes on the setup process (development, testing, production) have been updated in the proper documentation

## Observations

- This PR is intentionally test-focused; production behavior was not changed.
- Some tests were stabilized by improving mocks and async handling under coverage runs.
- A few existing production issues were only documented with `FIXME` comments so they can be addressed later in a dedicated functional change.

## Please, review

- Test intent and coverage quality, especially around:
  - generation dashboard table interactions
  - generation form edge cases
  - service-layer request building and cancellation behavior
  - shared i18n test mock stability
- Whether any test is over-coupled to current implementation details.

## How to check the new features

- Run the updated focused specs individually if needed.
- Run the coverage command and verify the touched targets pass under coverage.
- Confirm there are no production behavior changes beyond test support and inline FIXME documentation.

## Deploy notes

- No deploy or configuration changes required.

### CODE REVIEW

**Review by dukenuken09 — COMMENTED** [review]
At first glance everything seems fine. I don't see that the AI ​​does strange things

File: src/containers/HolderChange/StateCity.test.jsx
Comment (MariteSomEnergia) [inline src/containers/HolderChange/StateCity.test.jsx:1]: This component  will be replaced by new development. Maybe this tests are not necessary.

File: src/containers/HolderChange/StateCity.jsx
Comment (MariteSomEnergia) [inline src/containers/HolderChange/StateCity.jsx:1]: This component will be replaced by new development. Maybe this tests are not necessary.

File: src/containers/HolderChange/PersonalData.test.jsx
Comment (MariteSomEnergia) [inline src/containers/HolderChange/PersonalData.test.jsx:1]: This component will be replaced by new development. Maybe this tests are not necessary.

File: src/components/SomStepper/SomStepper.test.jsx
Comment (MariteSomEnergia) [inline src/components/SomStepper/SomStepper.test.jsx:1]: It is necessary this test when we use a component from somenergia-ui llibrary?

File: src/components/review/ReviewField.test.jsx [hunk 1]
```diff
@@ -1,8 +1,23 @@
...
+  render,
+  screen,
+} from "@testing-library/react"
+
+import SummaryContext, {
+  SummaryContextProvider,
+} from "../../context/SummaryContext"
 import ReviewField from "./ReviewField"
 
+const SummaryFieldObserver = () => {
```
Comment (MariteSomEnergia) [inline src/components/review/ReviewField.test.jsx:15]: I don't understand the utilitty

**Review by MariteSomEnergia — CHANGES_REQUESTED** [review]
I  think that to avoid future clean work it is better remove tests related with holderchange form which will be remove in a near future
